File: app/src/main/java/seveida/firetvforreddit/subreddit/domain/objects/ThreadMetadata.java
package seveida.firetvforreddit.subreddit.domain.objects;

import android.net.Uri;


import java.time.LocalDateTime;

import androidx.annotation.NonNull;

public class ThreadMetadata {

    @NonNull public final Integer threadId;
    @NonNull public final Uri imageUrl;
    @NonNull public final String title;
    @NonNull public final String op;
    @NonNull public final VoteCount voteCount;
    @NonNull public final LocalDateTime created;
    public final int commentCount;

    public ThreadMetadata(@NonNull Integer threadId, @NonNull Uri imageUrl,
                          @NonNull String title, @NonNull String op,
                          @NonNull VoteCount voteCount, @NonNull LocalDateTime created,
                          int commentCount) {
        this.threadId = threadId;
        this.imageUrl = imageUrl;
        this.title = title;
        this.op = op;
        this.voteCount = voteCount;
        this.created = created;
        this.commentCount = commentCount;
    }


}

File: domain_objects/src/main/java/baron/severin/domain_objects/ThreadMetadata.kt
package baron.severin.domain_objects

import android.net.Uri


import org.threeten.bp.Instant
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneId
import seveida.firetvforreddit.response.objects.Child
import seveida.firetvforreddit.response.objects.Data_

data class ThreadMetadata(val threadId: String, val previewImageUrl: Uri,
                          val title: String, val op: UserMetadata,
                          val voteCount: VoteCount, val created: LocalDateTime,
                          val commentCount: Int)

fun Child.toThreadMetadata(): ThreadMetadata {
    val data = this.data

    val id = data.id
    val imageUri = Uri.parse(data.thumbnail)
    val title = data.title

    val opName = data.author
    val opId = data.authorFullname
    val op = baron.severin.domain_objects.UserMetadata(opId, opName)

    val voteCount = baron.severin.domain_objects.VoteCount(data.ups, data.downs)

    val created = Instant.ofEpochMilli(data.created.toLong()).atZone(ZoneId.systemDefault()).toLocalDateTime()
    val commentCount = data.numComments
    return baron.severin.domain_objects.ThreadMetadata(id, imageUri, title, op, voteCount, created, commentCount)
}
